LLama.Unittest/ModelsParamsTests.cs

```diff
@@ -7,6 +8,8 @@ public class ModelsParamsTests
         [Fact]
         public void SerializeRoundTripSystemTextJson()
         {
+            var options = new JsonSerializerOptions();
```

Comment:
The `JsonConverter` attribute ([here](https://github.com/SciSharp/LLamaSharp/blob/d87d654a342179edb32bf85db26c6bec203567df/LLama/Common/ModelParams.cs#L96)) is meant to do this automatically. I'm not sure why that's not working.

Although this change fixes the test failure it doesn't really fix the problem - the `EncodingConverter` is an `internal` type so no users of LLamaSharp would actually be able to do this (and the intention is that serialization should "just work" without any extra config).

Reply:
You are right, I overlooked the fact of "internal".

After .NET 8, JsonSerializer has some [breaking change](https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/8.0/metadata-resolving) that caused custom JsonConverter on Encoding to become ineffective. You can refer to the [Ref above for more information.](https://github.com/dotnet/runtime/issues/95893#issuecomment-1851644867).

In this case, either convert EncodingConverter to public, but still need to remember to configure this converter during serialization. Or we need to find another way to serialize the Encoding property.

Reply:
Aha! Thank you, I've been looking through documentation trying to work out what was wrong for half an hour 🤦‍♂️

Reply:
I'll have a fiddle with trying other ways to do it

Reply:
@xbotter I've pushed up an alternative way to handle the `Encoding`. Eagerly converting it into a string, which can then be serialized directly.

Reply:
It looks good. ❤️

Reply:
Over on the issue you linked eiriktsarpalis suggested that all converters should be made public to support the source generator serialization ([ref](https://github.com/dotnet/runtime/issues/95893#issuecomment-1852344280)). Do you want to make that change in this PR too (for the `TensorSplitsCollectionConverter`)?

Reply:
Updated